reddit.js

```diff
@@ -2,4 +2,43 @@ var reddit = require("./data.json")
 
 // Write your solutions below
 
-console.log(reddit["data"]);
\ No newline at end of file
+//forEach iterator for all titles
+
+reddit["data"].children.forEach(function (title) {
+	console.log(title.data.title);
+});
+
+//map iterator for array of permalinks
+
+var permalinks = reddit["data"].children.map(function (link) {
+	return "http://reddit.com" + link.data.permalink;
+});
+
+console.log(permalinks);
+
+//filter iterator for actual text in selftext key
+
+var containsText =  function (text) {
+	return text.data.selftext != "";
+};
+
+var selfTextTrue = reddit["data"].children.filter(containsText);
```

Comment:
Again, you could use ```child``` here instead of ```containsText``` to more accurately represent what you're iterating over.